docs/argo-server-sso.md

```diff
@@ -188,6 +188,25 @@ sso:
   userInfoPath: /oauth2/v1/userinfo
 ```
 
+## Filter groups
+
+> v3.5 and after
+
+```yaml
+sso:
+    # Specify a list of regular expressions to filter the groups returned by the OIDC provider.
+    filterGroupsRegex:
+    - ".*argo-wf.*"
+    - ".*argo-workflow.*"
+```
+
+- `filterGroups` is a list of regular expressions that will be used to filter the groups returned by the OIDC provider.
+- This is an optional field. If not specified, no filtering will be done.
+- The regular expressions are case-sensitive.
+- If the list is not empty, only groups that match at least one of the regular expressions will be used in SSO RBAC.
+- Argo workflows uses cookies to store the groups' information. If the number of groups returned by OIDC provider is too large, the cookie size will exceed the  **4KB** limit and the login will fail. To avoid this, you can use `filterGroups` to filter out the groups that are not relevant to SSO RBAC.
+- Also, useful if you have multiple applications using the same OIDC provider, and you want to restrict access to only those groups that are relevant to Argo Workflows.
```

Comment:
- bullet points should only be used for lists. this should be in paragraph form unless a list is specified (usually with a colon (:)), consistent with the rest of the docs
- `filterGroupsRegex` is the name of the field
- "Argo Workflows" should always be in title case.
- grammar and consistency issues:
  - use "the OIDC provider" consistently.
- use present tense, [per k8s style guide](https://kubernetes.io/docs/contribute/style/style-guide/#use-present-tense)
- use active voice, [per k8s style guide](https://kubernetes.io/docs/contribute/style/style-guide/#use-active-voice)
- please use the [previous suggestions](https://github.com/argoproj/argo-workflows/pull/11778#discussion_r1320073919), such as for the link for the cookie limit and the wording and tense therein. 
  - no need to bold the 4KB
- typically the example use-cases come before the description of the feature in this page. i.e. the "motivation" for the feature comes first, and then how to use the feature

Reply:
I can rewrite this myself with the copy-edits above, but I am unsure if you will follow those this time